File: algo-linear-search/python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    for idx, elem in enumerate(array_to_search_through):
        if value_to_find == elem:
            return idx

def global_linear_search(value_to_find, array_to_search_through):
    result = []

    for idx, elem in enumerate(array_to_search_through):
        if value_to_find == elem:
            result.append(idx)
    
    return result

File: algo-linear-search/js/linearSearch.js
/**
 * @param  {} valueToFind
 * @param  {} arrayToSearchThrough
 */
exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
  let result
  let len = arrayToSearchThrough.length
  for (let i = 0; i < len; i++) {
      if (valueToFind === arrayToSearchThrough[i]) {
        result = i
      }
  }
  return result
}

exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {
    let result = []
    let len = arrayToSearchThrough.length
    for (let i = 0; i < len; i++) {
        if (valueToFind === arrayToSearchThrough[i]) {
          result.push(i)
        }
    }
    return result
}